spec/forms/state_file/taxes_owed_form_spec.rb

```diff
@@ -1,142 +1,172 @@
+        context "when the withdrawal date is after the payment deadline" do
+          let(:withdrawal_month) { (payment_deadline_date + 1.day).month }
+          let(:withdrawal_day) { (payment_deadline_date + 1.day).day }
+
+          it "is not valid" do
+            form = described_class.new(intake, params)
+            expect(form).not_to be_valid
+            expect(form.errors).to include :date_electronic_withdrawal
           end
+        end
+
+        context "when it is the very last minute before the payment deadline" do
+          let(:app_time) { payment_deadline_datetime - 1.minute }
+          let(:withdrawal_month) { (payment_deadline_date + 1.day).month }
+          let(:withdrawal_day) { (payment_deadline_date + 1.day).day }
 
-          let(:valid_params) do
-            {
-              date_electronic_withdrawal_month: '4',
-              date_electronic_withdrawal_year: (MultiTenantService.new(:statefile).current_tax_year + 1).to_s,
-              date_electronic_withdrawal_day: '30',
-              app_time: pre_deadline_withdrawal_time.to_s
-            }.merge(bank_info_params)
+          it "it rejects an invalid withdrawal date" do
+            form = described_class.new(intake, params)
+            expect(form).not_to be_valid
+            expect(form.errors).to include :date_electronic_withdrawal
           end
+        end
 
-          let!(:intake) {
-            create :state_file_md_intake,
-                   payment_or_deposit_type: "unfilled",
-                   withdraw_amount: withdraw_amount
-          }
+        context "when it is the very first minute on the payment deadline" do
+          let(:app_time) { payment_deadline_datetime }
+          let(:withdrawal_month) { (payment_deadline_date + 1.day).month }
+          let(:withdrawal_day) { (payment_deadline_date + 1.day).day }
 
-          it "updates the intake" do
-            form = described_class.new(intake, valid_params)
+          it "it ignores an invalid withdrawal date" do
+            form = described_class.new(intake, params)
             expect(form).to be_valid
-            form.save
+          end
+        end
 
-            intake.reload
-            expect(intake.payment_or_deposit_type).to eq "direct_deposit"
-            expect(intake.account_type).to eq "checking"
-            expect(intake.routing_number).to eq "019456124"
-            expect(intake.account_number).to eq "12345"
-            expect(intake.date_electronic_withdrawal).to eq Date.parse("April 30th, #{current_year}")
+        context "when the withdrawal date is in the past" do
+          let(:withdrawal_month) { (app_time - 1.days).month }
+          let(:withdrawal_day) { (app_time - 1.days).day }
+
+          it "is not valid" do
+            form = described_class.new(intake, params)
+            expect(form).not_to be_valid
+            expect(form.errors).to include :date_electronic_withdrawal
```

Comment:
should we be more explicit about what error is attached? for example, this one vs the one below it are maybe different errors & i'd expect to see that in the tests